After the recent switch to resteasy-reactive we are unable to use resteasy-classic or jersey jax-rs clients.

### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

dependencies

### Describe the bug

We have a number of custom SPI's which are using jax-rs client and more specifically currently Resteasy "classic". 
After the switch to resteasy-reactive we are unable to start Quarkus build due to various initialization issues and collisions.
* We tried to find and use resteasy-reactive but looks like the only one provided is in Quarkus which is and extensions and currently not enabled in Keycloak.
* We tried to switch the Jersey, but we still failed to build Quarkus image because of issues on the classloader. Looks like Jersey is not implementing some things but also we are getting same ServiceLoader issues ` Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
     [exec]     [error]: Build step io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor#setupEndpoints threw an exception: java.util.ServiceConfigurationError: jakarta.ws.rs.ext.RuntimeDelegate: org.glassfish.jersey.internal.RuntimeDelegateImpl not a subtype
     [exec]     at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:593)
     [exec]     at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1244)
     [exec]     at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1273)
     [exec]     at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1309)
     [exec]     at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1393)`

The options we have left is switch find completely standalone client or maybe keycloak can enable the resteasy-reactive extension. 


### Version

23.0.0

### Expected behavior

We expect to be able to use resteasy "classic" or reactive client in our extensions. 

### Actual behavior

Seem we can't find a way to augment Quarkus once we add our SPI's 

### How to Reproduce?

Currently, we do not have small example, but we can try to provide one if needed.

### Anything else?

We understand it might not be a good idea for Keycloak to enable a particular Quarkus extensions. We are looking for other ideas or suggestions as well.

--- a/docs/documentation/release_notes/topics/23_0_0.adoc
+++ b/docs/documentation/release_notes/topics/23_0_0.adoc
@@ -23,3 +23,7 @@ Therefore, users of Keycloak can do passkey registration and authentication by e
 
 Both synced passkeys and device-bound passkeys can be used for both Same-Device and Cross-Device Authentication.
 However, passkeys operations success depends on the user's environment. Make sure which operations can succeed in https://passkeys.dev/device-support/[the environment].
+
+= RESTEasy Reactive
+
+Keycloak has switched to RESTEasy Reactive. Applications using `quarkus-resteasy-reactive` should still benefit from a better startup time, runtime performance, and memory footprint, even though not using reactive style/semantics. SPI's that depend directly on JAX-RS API should be compatible with this change. SPI's that depend on RESTEasy Classic including `ResteasyClientBuilder` will not be compatible and will require update, this will also be true for other implementation of the JAX-RS API like Jersey.

--- a/docs/documentation/upgrading/topics/keycloak/changes-23_0_0.adoc
+++ b/docs/documentation/upgrading/topics/keycloak/changes-23_0_0.adoc
@@ -135,6 +135,9 @@ Stream<GroupModel> getTopLevelGroupsStream(RealmModel realm,
 
 Endpoint `GET {keycloak server}/realms/{realm}/groups/{group_id}/children` added as a way to get subgroups of specific groups that support pagination
 
+= RESTEeasy Reactive
+Relying on RESTEasy Classic is not longer an option because it is not available anymore. Migration will be needed for SPI's and code that is relying on RESTEasy Classic and related packages part of  `org.jboss.resteasy.spi.*`.
+
 = Partial export requires manage-realm permission
 
 The endpoint `POST {keycloak server}/realms/{realm}/partial-export` and the corresponding action in the admin console now require `manage-realm` permission for execution instead of `view-realm`. This endpoint exports the realm configuration into a JSON file and the new permission is more appropriate. The parameters `exportGroupsAndRoles` and `exportClients`, which include the realm groups/roles and clients in the export respectively, continue managing the same permissions (`query-groups` and `view-clients`).

